### PR TITLE
Add table climbing

### PR DESCRIPTION
**Pull request by DamianX**
The long awaited feature.

:cl:
 * rscadd: Added table climbing.

### CODE REVIEW

**Comment by kilozombie** [comment]
please no
edit: wait it's a meme PLEASE YES

**Comment by Probe1** [comment]
You're not adding this without making significant revisions to every map to prevent people from waltzing into departments.

**Comment by kilozombie** [comment]
Check that code

**Comment by Intigracy** [comment]
10/10 code

**Comment by Probe1** [comment]
We've had enough meme PRs this season.  Label your PRs for what they do or don't make them at all please.

**Comment by Exxion** [comment]
Quick someone post NO FUN ALLOWED

**Comment by Probe1** [comment]
z7herxu

**Comment by Exxion** [comment]
Yeah that one

**Comment by troots** [comment]
dont be a probe and read the code

**Comment by Probe1** [comment]
the sad truth is I read it but doubted myself and thought I didn't understand it.  so I went back to leave a comment and re-read the code while waking up and see if I could figure out how it worked.

turned out I was just dumb.

**Comment by Exxion** [comment]
If you had opened this on April Fools', it would have been merged immediately, albeit only for a day

**Comment by Intigracy** [comment]
Nah this is worth keeping around, it'd stop the adminhelps asking how to tableclimb.

**Comment by Exxion** [comment]
Oh yeah, due to the sound it would also remove the use of a bar deskbell
Ship it

**Comment by ihadtoregisterforthis** [comment]
>4 days into the season 
>We've had enough meme PRs this season

**Comment by Intigracy** [comment]
I'll merge this tomorrow.

**Comment by kilozombie** [comment]
yis

File: code/game/objects/structures/tables_racks.dm [hunk 1]
```diff
@@ -615,6 +615,18 @@
...
+	if(O != user || !ishuman(usr) || !in_range(src, usr) || usr.incapacitated() || usr.lying) // Doesn't work if you're not dragging yourself, not a human, not in range or incapacitated
+		return
+	var/mob/living/carbon/M = user
+	M.apply_damage(2, BRUTE, LIMB_HEAD, used_weapon = "[src]")
+	M.Knockdown(1)
+	if (prob(50))
+		playsound(M, 'sound/items/trayhit1.ogg', 50, 1)
+	else
+		playsound(M, 'sound/items/trayhit2.ogg', 50, 1)
+	M.visible_message("<span class='danger'>You bang your head on \the [src.name].</span>", "<span class='danger'>[usr] bangs his head on \the [src.name].</span>", "You hear a bang.")
```
Comment (Intigracy) [inline code/game/objects/structures/tables_racks.dm]: change all the `[src.name]` to `[src]`

`bangs his head` needs to be `bangs \his head`

File: code/game/objects/structures/tables_racks.dm [hunk 2]
```diff
@@ -615,6 +615,18 @@
 		return
 	return ..()
 
+/obj/structure/table/MouseDrop_T(atom/movable/O, mob/user)
+	if(O != user || !ishuman(usr) || !in_range(src, usr) || usr.incapacitated() || usr.lying) // Doesn't work if you're not dragging yourself, not a human, not in range or incapacitated
```
Comment (Intigracy) [inline code/game/objects/structures/tables_racks.dm]: `!in_range(src, usr)` should be `!Adjacent(user)`

replace usr with user for these, usr isn't very sane

**Review by Intigracy — CHANGES_REQUESTED** [review]
few small things

File: code/game/objects/structures/tables_racks.dm (hunk 1 above)
Comment (Exxion) [reply code/game/objects/structures/tables_racks.dm]: Also another `usr`

**Comment by DamianX** [comment]
Oh fuck the madman reopened it

**Comment by SonixApache** [comment]
proper title when

**Comment by Exxion** [comment]
You bungled it
He said change them to `[src]`, not `[name]`

**Comment by Exxion** [comment]
@SonixApache Never

1. Get a calendar
2. Shut up

**Comment by DamianX** [comment]
It doesn't NOT work though, I wonder if there's a difference.

File: code/game/objects/structures/tables_racks.dm [hunk 3]
```diff
@@ -615,6 +615,18 @@
...
+	if(O != user || !ishuman(user) || !Adjacent(src, user) || user.incapacitated() || user.lying) // Doesn't work if you're not dragging yourself, not a human, not in range or incapacitated
+		return
+	var/mob/living/carbon/M = user
+	M.apply_damage(2, BRUTE, LIMB_HEAD, used_weapon = "[src]")
+	M.Knockdown(1)
+	if (prob(50))
+		playsound(M, 'sound/items/trayhit1.ogg', 50, 1)
+	else
+		playsound(M, 'sound/items/trayhit2.ogg', 50, 1)
+	M.visible_message("<span class='danger'>You bang your head on \the [name].</span>", "<span class='danger'>[user] bangs \his head on \the [name].</span>", "You hear a bang.")
```
Comment (Intigracy) [inline code/game/objects/structures/tables_racks.dm]: `>[name]`

lmao I said `[src]`

**Review by Intigracy — CHANGES_REQUESTED** [review]

**Comment by Intigracy** [comment]
Just a standard we apply.

Doing [src] or [user] or [M] automatically allows reference of gender (with \his) and a few other things

**Review by Intigracy — APPROVED** [review]

**Comment by Exxion** [comment]
It DOES not work, actually.
If `src` were a proper noun, it would not appear properly if you used `[name]`

**Comment by Intigracy** [comment]
this shouldn't ever get reverted

**Comment by kernikov** [comment]
>doesn't apply brain damage

smh

**Comment by DamianX** [comment]
You think I didn't think about it?

**Pull request merged**